src/python/pants/build_graph/target.py

```diff
@@ -139,9 +141,47 @@ def check_unknown(self, target, kwargs, payload):
           args=''.join('\n  {} = {}'.format(key, value) for key, value in unknown_args.items())
         ))
 
+  class TagAssignments(Subsystem):
+    """Tags to be applied to targets defined in a single source file."""
+
+    options_scope = 'target-tag-assignments'
+
+    @classmethod
+    def register_options(cls, register):
+      register('--tag-targets-file', type=str, default=None, fingerprint=True,
+               help='Source JSON file with tag assignments for targets. Ex: \
```

Comment:
Rather than JSON (which doesn't support comments), it might be preferable have this be a `type=dict` option, marked `fromfile=True`.

That way folks could define the value either directly inline in `pants.ini`, or in a separate file via `@fromfile`.

Apparently `fromfile=True` isn't documented here: https://www.pantsbuild.org/options.html#dict-options ... but dict options are at least.

Reply:
Sounds good - I started down that path and got hung up on getting the `fromfile` option under tests, will push an update once I get that resolved

Reply:
I added d4e53c7 without testing the `fromfile` option explicitly as there are other tests to show that `fromfile` works and can come back around to adding coverage if needed.